engine/src/main/java/org/terasology/telemetry/metrics/Metric.java

```diff
@@ -21,24 +21,43 @@
 import org.slf4j.LoggerFactory;
 import org.terasology.context.Context;
 import org.terasology.engine.subsystem.DisplayDevice;
+import org.terasology.module.sandbox.API;
 import org.terasology.registry.CoreRegistry;
+import org.terasology.telemetry.Metrics;
+import org.terasology.telemetry.TelemetryCategory;
 import org.terasology.telemetry.TelemetryField;
 
 import java.lang.reflect.Field;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
 /**
  * An abstract class for a telemetry metric.
- * A new metric should extends this class, with annotation {@link org.terasology.telemetry.TelemetryCategory} and all metric fields with annotation {@link org.terasology.telemetry.TelemetryField}.
+ * A new metric should extends this class, with annotation {@link org.terasology.telemetry.TelemetryCategory}.
+ * All metric fields should be annotated {@link org.terasology.telemetry.TelemetryField}.
  * An example is {@link org.terasology.telemetry.metrics.SystemContextMetric}.
  */
+@API
 public abstract class Metric {
 
     private static final Logger logger = LoggerFactory.getLogger(Metric.class);
 
-    protected Map metricMap;
+    protected Map metricMap = new HashMap();
+
+    /**
+     * One a new metric is created, it should be added to the {@link org.terasology.telemetry.Metrics}.
+     * Use this constructor (super(context);) will add metric automatically to Metrics class.
+     * @param context the game context.
+     */
+    public Metric(Context context) {
+        Metrics metrics = context.get(Metrics.class);
+        addToMetrics(metrics);
```

Comment:
I don't like this kind of API for two reasons:
1. The initialization in the Metrics class looks like floating and unused object instances and it is not clear where a instance to a metric is stored.
Also, you do the initialization in metrics, so Metrics schould know where to store a metric.
2. You can't use a metric in any other place without a valid context containing a Metrics class. There is no way to create an instance e.g. for tests.

If you want to force metrics to be on the metrics class and inside the context, why not use an annotation as marker and auto register all annotated metrics. This would also drive the Open-Closed Principle as you could add extra Metrics without changing the Metrics class again and again for each metric.